### ISSUE TITLE
Test the SDK in OSGI and Java9

### ISSUE THREAD

**Issue by shorea** [aws/aws-sdk-java-v2#59]
Make sure we play nice with the module systems in OSGI and Java9 (especially our detection of the default HTTP implementation). Ensure that we can provide modularized service clients to OSGI environments instead of a single uber jar containing the entire SDK.

**Comment by sindremehus** [comment]
For Java 9 compatibility, at the very least please specify a stable "Automatic-Module-Name" in the jar manifests, and make sure there are no split packages across the different artifacts.

**Comment by jenslauterbach** [comment]
We use DynamoDB SDK 1.x in a OSGi container and have to build our own bundle. Would be great if the new SDK would be OSGi compatible from the start.

**Comment by bennyflint** [comment]
Any progress or info here. The split packages makes modularizing my app effectively impossible.

**Comment by dagnir** [comment]
@bennyflint Yes, there's been a few PR's merged that hopefully address some of these issues:

 - #475 
 - #480 
 - #526

**Comment by dagnir** [comment]
#584

**Comment by pegr69** [comment]
What is going on with this issue, is there any progress into getting OSGi supported in clientbuilder code?

**Comment by aalmiray** [comment]
If split packages are no longer an issue (I sure hope so) then adding support for Java 9+ modules may still be done using the [ModiTect](https://github.com/moditect/moditect) plugin. This way the codebase remains Java 8 compatible. The build would have to jump to Java 11 at a minimum if it's not done so already.
